### ISSUE TITLE
freerdp crashed when using copy paste

### ISSUE THREAD

**Issue by adub30** [FreeRDP/FreeRDP#4344]
So the issue I'm having is freerdp crashing when using copy  & paste in remote session, same for mouse copy paste.

running linux mint 18.3 
freerdp-x11 2.0

script used
xfreerdp /multimon:force +clipboard /sound /microphone:sys:alsa /sec:rdp /d:contoso /v:username /v:192.xxx.xxx.xxx  

I RDP into win server 2012, we just uprated the server from win server 2008 and had zero issues with copy & paste being used. 

*note linx mint is installed on slim-clients, in a mixed environment some slims are win 10 and the issue does not happen, also it only crashes while in office 2016 product, word, excel, outlook, any ideas?

**Comment by akallabeth** [comment]
@adub30 Some additional information that could help:
* The output of ```xfreerdp /buildconfig```
* Do you use self compiled or distro package?
* ```/log-level:trace``` (very long, but may contain some hint on what is actually going on)
* If self compiled, compile with ```-DWITH_SANITIZE_ADDRESS=ON``` for a stack trace on crash

**Comment by akallabeth** [comment]
@adub30 Any update? Not reproducible here, so some input is required for further debugging.

**Comment by adub30** [comment]
I ended up re-installing Linux minus remmina, no more crashes, thank you for your follow up.

Aaron Windhorn/ IT Support / Network Engineer
Merchants Acceptance Corp.
Phone: 866.883.2392 ext. 7440
Cell: 206.250.3562 | | Fax: 888.278.0404
1314 Auburn Way NO. Auburn Wa 98002



.By responding or taking action upon this email you are consenting to ongoing electronic mail communication
including but not limited to statements, terms of agreement, and account details.
CONFIDENTIALITY NOTE: This electronic message is intended only for the use of the individual(s) or entity named
above and may contain confidential, legally privileged or proprietary information which is not for general use or
disclosure. If you are not the intended recipient, you are notified that any disclosure, copying, distribution or the taking
of any action in reliance on the contents of this message is strictly prohibited. If you receive this message in error,
please alert me by reply message and then delete this message from your system. Your cooperation is appreciated.

**Comment by akallabeth** [comment]
Closing as not reproducible then.

**Comment by soerendip** [comment]
I have the same issue. Running on Ubuntu 18.04.3.

**Comment by Erbureth** [comment]
Hi, I can reproduce the issue consistently in IntelliJ Idea Community Edition:
In Vim mode, yank multiple lines, then paste them somewhere else. FreeRDP crashes.

[xfreerdp.log](https://github.com/FreeRDP/FreeRDP/files/4388696/xfreerdp.log)

[xfreerdp.buildconfig.txt](https://github.com/FreeRDP/FreeRDP/files/4388677/xfreerdp.buildconfig.txt)

**Comment by cjsin** [comment]
I am seeing this with:

freerdp client version:
  debian client package     freerdp2-x11 2.0.0~git20190204.1.2693389a+dfsg1-1+deb10u1 amd64  

xrdp server version:
   fedora server package xrdp-0.9.13-1.fc30.x86_64

While it is not clear whether the bug is in the client or the server, it is the client that crashes - the server remains running and can be connected to again.

Relevant output from /log-level:trace

[00:56:15:335] [13064:13070] [ERROR][com.freerdp.channels.cliprdr.client] - Invalid short format list length: 24
[00:56:15:335] [13064:13070] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_process_format_list failed with error 1359!
[00:56:15:335] [13064:13070] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_order_recv failed with error 1359!
[00:56:15:336] [13064:13065] [ERROR][com.freerdp.core] - cliprdr_virtual_channel_client_thread reported an error. Error was 1359
[00:56:15:336] [13064:13065] [ERROR][com.freerdp.core] - checkChannelErrorEvent() failed - 0
[00:56:15:336] [13064:13065] [INFO][com.freerdp.client.common] - Network disconnect!
[00:56:15:336] [13064:13065] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor


In my situation, (linux xrdp desktop server):
   - the errors happened on a 'copy' (right-click -> Copy in a popup menu) . ie a copy to what would be the SECONDARY clipboard in X11
   - the errors occurred in multiple applications (chromium, thunderbird, a terminal app "Terminator") 
   - copy/paste of the PRIMARY selection within the freerdp window did not cause a crash (although this data was not propagated into the environment of the client)
 
Additionally, I tested whether it was a complex data format that was triggering the bug, (such as html in chromium for example). However I believe the error happens for plain text. This was the command used (in the linux server side):

$>   date | xsel -i --clipboard

which is only setting the clipboard value to basic text, but immediately crashes the client


That would seem to indicate that the client is failing 
with errors "Invalid short format list length: 24"
and "cliprdr_process_format_list failed with error 1359!"
even with simple text values.

**Comment by cjsin** [comment]
Additional xfreerdp /buildconfig information:

This is FreeRDP version 2.0.0-dev5 (2693389a+debian)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_JOURNALD_H=TRUE HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 HAVE_XI_TOUCH_CLASS=1 WITH_ALSA=ON WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=ON WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=OFF WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=TRUE WITH_FFMPEG=TRUE WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=ON WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=ON WITH_ICU=ON WITH_IPP=OFF WITH_JPEG=ON WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_MACAUDIO=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PAM=ON WITH_PCSC=ON WITH_PROFILER=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=ON WITH_SERVER_CHANNELS=ON WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=ON WITH_THIRD_PARTY=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=ON WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XDAMAGE=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XKBFILE=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XSHM=ON WITH_XTEST=ON WITH_XV=ON WITH_ZLIB=ON
Build type:          RELWITHDEBINFO
CFLAGS:              -g -O2 -fdebug-prefix-map=/build/freerdp2-Bj0zA6/freerdp2-2.0.0~git20190204.1.2693389a+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -g -DWINPR_DLL
Compiler:            GNU, 8.2.0
Target architecture: x64

**Comment by cjsin** [comment]
May be fixed already under issue #4879 
However that fix was in version 2.0.0.
The bug is present in my debian 2.0.0-dev5 package.
Unclear to me whether a debian 2.0.0  'dev5' package is actually 2.0.0, or an alpha/beta version

**Comment by wsdookadr** [comment]
Versions used:

| software | version | comments | 
|--|--|--|
| [1] xrdp | 0.9.12-1.1 from debian bullseye RC1 | |
| [2] freerdp-nightly | 3.0.0+0\~20210723014817.993~1.gbp2c31c3 | [nightly build #994 for debian buster](https://ci.freerdp.com/job/freerdp-nightly-binaries/architecture=amd64,distribution=buster,label=pkg-deb/994/) |
| [3] freerdp2-x11 | 2.0.0~git20190204.1.2693389a+dfsg1-1+deb10u2 | package from the official debian buster repository |

While [2] and [3] act as clients (they're both providing xfreerdp just in different versions), [1] is the server.

Steps to reproduce: When presented with the xrdp login screen, select Xvnc for the session, write the username and password. Then start Firefox, select some text and press Copy and xfreerdp crashes with this error:

```
user@home:/tmp$ xfreerdp /v:192.168.1.111 +glyph-cache /relax-order-checks /u:work /size:"90%" /clipboard +clipboard
[10:44:55:017] [20242:20243] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[10:44:55:018] [20242:20243] [INFO][com.freerdp.client.x11] - Property 387 does not exist
[10:44:55:048] [20242:20243] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[10:44:55:048] [20242:20243] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_RGB16
[10:44:55:058] [20242:20243] [INFO][com.winpr.clipboard] - initialized POSIX local file subsystem
[10:45:06:005] [20242:20248] [ERROR][com.freerdp.channels.cliprdr.client] - Invalid short format list length: 24
[10:45:06:005] [20242:20248] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_process_format_list failed with error 1359!
[10:45:06:005] [20242:20248] [ERROR][com.freerdp.channels.cliprdr.client] - cliprdr_order_recv failed with error 1359!
[10:45:06:005] [20242:20243] [ERROR][com.freerdp.core] - cliprdr_virtual_channel_client_thread reported an error. Error was 1359
[10:45:06:006] [20242:20243] [ERROR][com.freerdp.core] - checkChannelErrorEvent() failed - 0
[10:45:06:006] [20242:20243] [INFO][com.freerdp.client.common] - Network disconnect!
[10:45:06:006] [20242:20243] [ERROR][com.freerdp.client.x11] - Failed to check FreeRDP file descriptor
```

The bug is present in [3] but does not manifest in [2].

Note: Found [out about](https://github.com/neutrinolabs/xrdp/issues/386#issuecomment-288168075) `autorun=Xvnc` xrdp setting along the way.
